app/models/user.php

```diff
@@ -82,5 +81,136 @@ public function login(LoginInfo $login_info)
             throw new RecordNotFoundException('No Record Found');
         }
         Session::set('username', $this->username);
+        Session::set('id', $row['id']);
+    }
+
+    public static function getAll()
+    {
+        $user = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM user WHERE id = ?", array(Session::get('id')));
+
+        foreach ($rows as $row) {
+            $user[] = new self($row);
+        }
+        return $user;
+    }
+
+    public static function getComments()
+    {
+        $comments = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM comment 
+                    WHERE username = ?", array(Session::get('username')));
+
+        foreach ($rows as $row) {
+            $comments[] = new self($row);
+        }
+        return $comments;
+    }
+
+    public function getId()
+    {
+        $db = DB::conn();
+        $id = $db->row("SELECT id FROM user WHERE id = ?", array(Session::get('id')));
+        $this->id = $id['id'];
+    }
+
+    public function updateProfile()
+    {
+        $params = array();
+        $temp = array('firstname' => $this->firstname,
+                    'lastname' => $this->lastname,
+                    'email' => $this->email,
+        );
+
+        foreach ($temp as $k => $v) {
+            if (!empty($v)) {
+                $params[$k] = $v;
+            }
+        }
+
+        if (!empty($params)) {
+            try {
+                $db = DB::conn();
+                $db->begin();
+                $db->update('user', $params, array('id' => $this->id));
+                $db->commit();
+            } catch (Exception $e) {
+                $db->rollback();
+                throw $e;
+            }
+        }
+    }
+
+    public function updatePassword()
+    {
+        if (!empty($this->newpassword) && !empty($this->cnewpassword)) {
+            $params = array('password' => md5($this->newpassword),
+                        'validate_password' => md5($this->cnewpassword),
+            );
+            $db = DB::conn();
+            $row = $db->row('SELECT password FROM user WHERE id = ?', array(Session::get('id')));
+
+            if ($row['password'] === md5($this->password) && $this->newpassword === $this->cnewpassword) {
+                 try {
+                    $db->begin();
+                    $db->update('user', $params, array('id' => $this->id));
+                    $db->commit();
+                } catch(Exception $e) {
+                    $db->rollback();
+                    throw $e;
+                }
+            } else {
+                throw new ValidationException('Password Mismatch');
+            }
+        } else {
+                throw new ValidationException('Password Mismatch');
+        }
+    }
+
+    public static function getUserInfo($user_id)
```

Comment:
I think the name and the return is not match? 
i suppose getUserInfo returns the user data with the id $user_id
but the return is array of user objects?